### PR TITLE
Bugs with console fixed.

### PR DESCRIPTION
**Pull request by kajojify**
- Clicking on tab head focuses a tab now. #2676 fixed.
- We can't click on different stuff in not focused tab by pressing left button of mouse now. #2738 fixed.

### CODE REVIEW

**Comment by mhils** [comment]
This looks great, thanks. How do you feel about 299df48 as an alternative fix for #2738? This would retain the ability to switch tabs, and would also allow us to have e.g. the command list in the right view and a grideditor on the left.

**Comment by kajojify** [comment]
@mhils Your fix looks more naturally and fixes #2738 :)
But it doesn't deal with a problem we discussed here https://github.com/mitmproxy/mitmproxy/issues/2676#issuecomment-354612811
We can still modify stuff in not focused tab.
For example:
screenshot from 2018-01-04 15-14-27
When being focused on **Command Reference** I can click on **Flow Details** by pressing left button of mouse and start interacting with it (without changing the colour of the header). Nevertheless all global keybindings still work only for tab with highlighted header. 
It looks a little bit strange, but if it isn't cosidered as an issue - ok.

**Comment by mhils** [comment]
Right, ideally we should change the focused pane to wherever the user clicked. How about we get rid of StackWidgetHeader/StackWidgetBody and just implement `StackWidget.mouse_event` to always change focus to the pane that the user clicked on?

File: mitmproxy/tools/console/window.py [hunk 1]
```diff
@@ -16,7 +16,10 @@
 
 
 class StackWidget(urwid.Frame):
-    def __init__(self, widget, title, focus):
+    def __init__(self, window, widget, title, focus):
+        self.f = focus
```
Comment (mhils) [inline mitmproxy/tools/console/window.py]: This needs a more self-explaining name. Maybe `.is_focused`?

File: mitmproxy/tools/console/window.py [hunk 2]
```diff
@@ -29,6 +32,14 @@ def __init__(self, widget, title, focus):
             header=header
         )
 
+    def mouse_event(self, size, event, button, col, row, focus):
+        if event == "mouse press" and button == 1:
+            if not self.f:
+                self.window.switch()
+                return True
+            else:
+                super().mouse_event(size, event, button, col, row, focus)
```
Comment (mhils) [inline mitmproxy/tools/console/window.py]: `return super()...`

**Comment by kajojify** [comment]
Talking about a thing, you described. It doesn't activate underlying action - just switching.

**Comment by mhils** [comment]
Right. My question was if it would be more intuitive to also activate the underlying action, like e.g. Chrome does it.

**Comment by kajojify** [comment]
@mhils Yes, it would be more intuitive, I think. So, do I need to include this feature into this PR or into next?

**Comment by kajojify** [comment]
Sorry for making things more complex :) I will try not to do that again.

**Pull request merged**